app/src/app/components/contents/common/FilterByName.tsx

```diff
@@ -0,0 +1,53 @@
+import debounce from "lodash.debounce";
+import { Dispatch, SetStateAction, useCallback, useEffect, useRef } from "react";
+import { Input } from "../../Base/Input";
+import { Button } from "../../Base/Button";
+import { X } from "lucide-react";
+
+interface FilterByNameProps {
+  setFilterByName: Dispatch<SetStateAction<string>>;
+  setPageNumber?: Dispatch<SetStateAction<number>>;
+}
```

Comment:
Actually, this is more generic than `FilterByName` - doesn't have to be name that you're filtering by, it could be anything, as interpreted by the action. Placeholder could be a prop, and this could just be `FilterByUserInput` or something. That's almost certainly going to happen when we introduce more complex filtering.

Reply:
great idea!! have passed in placeholder text and named it `FilterInput`